firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/internal/util/conversions.kt

```diff
@@ -115,259 +36,6 @@ internal fun makeMissingCaseException(source: String, ordinal: Int): Serializati
-    HarmCategory.HATE_SPEECH -> com.google.firebase.vertexai.common.shared.HarmCategory.HATE_SPEECH
-    HarmCategory.SEXUALLY_EXPLICIT ->
-      com.google.firebase.vertexai.common.shared.HarmCategory.SEXUALLY_EXPLICIT
-    HarmCategory.DANGEROUS_CONTENT ->
-      com.google.firebase.vertexai.common.shared.HarmCategory.DANGEROUS_CONTENT
-    HarmCategory.CIVIC_INTEGRITY ->
-      com.google.firebase.vertexai.common.shared.HarmCategory.CIVIC_INTEGRITY
-    HarmCategory.UNKNOWN -> com.google.firebase.vertexai.common.shared.HarmCategory.UNKNOWN
-    else -> throw makeMissingCaseException("HarmCategory", ordinal)
-  }
-
-internal fun HarmBlockMethod.toInternal() =
-  when (this) {
-    HarmBlockMethod.SEVERITY -> com.google.firebase.vertexai.common.shared.HarmBlockMethod.SEVERITY
-    HarmBlockMethod.PROBABILITY ->
-      com.google.firebase.vertexai.common.shared.HarmBlockMethod.PROBABILITY
-    else -> throw makeMissingCaseException("HarmBlockMethod", ordinal)
-  }
-
-internal fun ToolConfig.toInternal() =
-  com.google.firebase.vertexai.common.client.ToolConfig(
-    functionCallingConfig?.let {
-      com.google.firebase.vertexai.common.client.FunctionCallingConfig(
-        when (it.mode) {
-          FunctionCallingConfig.Mode.ANY ->
-            com.google.firebase.vertexai.common.client.FunctionCallingConfig.Mode.ANY
-          FunctionCallingConfig.Mode.AUTO ->
-            com.google.firebase.vertexai.common.client.FunctionCallingConfig.Mode.AUTO
-          FunctionCallingConfig.Mode.NONE ->
-            com.google.firebase.vertexai.common.client.FunctionCallingConfig.Mode.NONE
-        },
-        it.allowedFunctionNames
-      )
-    }
-  )
-
-internal fun HarmBlockThreshold.toInternal() =
-  when (this) {
-    HarmBlockThreshold.NONE ->
-      com.google.firebase.vertexai.common.shared.HarmBlockThreshold.BLOCK_NONE
-    HarmBlockThreshold.ONLY_HIGH ->
-      com.google.firebase.vertexai.common.shared.HarmBlockThreshold.BLOCK_ONLY_HIGH
-    HarmBlockThreshold.MEDIUM_AND_ABOVE ->
-      com.google.firebase.vertexai.common.shared.HarmBlockThreshold.BLOCK_MEDIUM_AND_ABOVE
-    HarmBlockThreshold.LOW_AND_ABOVE ->
-      com.google.firebase.vertexai.common.shared.HarmBlockThreshold.BLOCK_LOW_AND_ABOVE
-    else -> throw makeMissingCaseException("HarmBlockThreshold", ordinal)
-  }
-
-internal fun Tool.toInternal() =
-  com.google.firebase.vertexai.common.client.Tool(
-    functionDeclarations?.map { it.toInternal() } ?: emptyList()
-  )
-
-internal fun FunctionDeclaration.toInternal() =
-  com.google.firebase.vertexai.common.client.FunctionDeclaration(name, "", schema.toInternal())
-
-internal fun com.google.firebase.vertexai.type.Schema.toInternal(): Schema =
-  Schema(
-    type,
-    description,
-    format,
-    nullable,
-    enum,
-    properties?.mapValues { it.value.toInternal() },
-    required,
-    items?.toInternal(),
-  )
-
 internal fun JSONObject.toInternal() = Json.decodeFromString<JsonObject>(toString())
```

Comment:
Could this be moved somewhere else?

Otherwise we may rename this file since "conversions" doesn't necessarily match

Reply:
Moved it to the Type file where other serialization stuff lives outside of common so we can delete conversions.kt